tests/testSpanSets.cc

```diff
@@ -0,0 +1,638 @@
+        BOOST_CHECK(spn.getY() == yStart);
+        BOOST_CHECK(spn.getMinX() == 0);
+        BOOST_CHECK(spn.getMaxX() == 4);
+        ++yStart;
+    }
+
+    // Test with a null SpanSet
+    afwGeom::SpanSet nullSpanSet;
+    auto nullAsOther = firstSS->intersectNot(nullSpanSet);
+    // A SpanSet intersectedNot with a null SpanSet should just return itself
+    BOOST_CHECK(*firstSS == *nullAsOther);
+
+    // A null SpanSet intersectedNot with anything should return a null SpanSet
+    auto spanSetAsOther = nullSpanSet.intersectNot(*firstSS);
+    BOOST_CHECK(nullSpanSet == *spanSetAsOther);
+}
+
+BOOST_AUTO_TEST_CASE(SpanSet_testUnion) {
+    auto ret = makeOverlapSpanSets();
+    auto firstSS = ret.first;
+    auto secondSS = ret.second;
+
+    auto overlap = firstSS->union_(*secondSS);
+
+    // Verify both SpanSets are combined together
+    int yStart = 0;
+    for (auto const & spn : *overlap) {
+        BOOST_CHECK(spn.getY() == yStart);
+        BOOST_CHECK(spn.getMinX() == 0);
+        BOOST_CHECK(spn.getMaxX() == 4);
+        ++yStart;
+    }
+
+    // Test with a null SpanSet
+    afwGeom::SpanSet nullSpanSet;
+    auto nullAsOther = firstSS->union_(nullSpanSet);
+    // A null SpanSet unioned with another SpanSet should return the other SpanSet
+    BOOST_CHECK(*nullAsOther == *firstSS);
+
+    auto spanSetAsOther = nullSpanSet.union_(*firstSS);
+    BOOST_CHECK(*spanSetAsOther == *firstSS);
+
+}
+
+BOOST_AUTO_TEST_CASE(SpanSet_testEquality) {
+    auto ret = makeOverlapSpanSets();
+    auto firstSS = ret.first;
+    auto secondSS = ret.second;
+    auto secondSSShift = secondSS->shiftedBy(0, 2);
+
+    BOOST_CHECK((*firstSS == *secondSS) == false);
+    BOOST_CHECK((*firstSS != *secondSS) == true);
+    BOOST_CHECK((*firstSS == *secondSSShift) == true);
+    BOOST_CHECK((*firstSS != *secondSSShift) == false);
+}
+
+BOOST_AUTO_TEST_CASE(SpanSet_testFunctor) {
+    // Test the remaining types of functors. Above code has tested ndarray functors
+    // need to test, constants, iterators, Images
+    namespace afwImage = lsst::afw::image;
+    int imageDim = 6;
+    int initialValue = 0;
+    afwImage::Image<int> imageObject(imageDim, imageDim, initialValue);
+    std::vector<int> vecObject(imageDim*imageDim, initialValue);
+
+    int spanRadius = 2;
+    auto SSShape = afwGeom::SpanSet::spanSetFromShape(spanRadius,
+                                                      afwGeom::Stencil::BOX)->shiftedBy(spanRadius,
+                                                                                       spanRadius);
+    // use a constant as a test of constantGetter
+    int dataValue = 6;
+    SSShape->applyFunctor([](afwGeom::Point2I pt, afwImage::Image<int>::Pixel & out, int in){ out = in; },
+                            imageObject, dataValue);
+    SSShape->applyFunctor([](afwGeom::Point2I, int & out, int in){out = in;},
+                             vecObject.begin(), dataValue);
+
+    // Check on the point input of a functor
```

Comment:
Consider moving this block to after the dataValue/initialValue checks, so that related code is together.